src/discovery.jl

```diff
@@ -269,13 +278,26 @@ function find_toolchain(toolkit_path, toolkit_version=find_toolkit_version(toolk
         host_compiler, host_version = gcc_possibilities[1]
     elseif Compat.Sys.iswindows()
         # Windows: just use cl.exe
-        vc_versions = ["VS140COMNTOOLS", "VS120COMNTOOLS", "VS110COMNTOOLS", "VS100COMNTOOLS"]
-        !any(x -> haskey(ENV, x), vc_versions) && error("Compatible Visual Studio installation cannot be found; Visual Studio 2015, 2013, 2012, or 2010 is required.")
-        vs_cmd_tools_dir = ENV[vc_versions[first(find(x -> haskey(ENV, x), vc_versions))]]
-        cl_path = joinpath(dirname(dirname(dirname(vs_cmd_tools_dir))), "VC", "bin", Sys.WORD_SIZE == 64 ? "amd64" : "", "cl.exe")
 
-        host_compiler = cl_path
-        host_version = nothing
+        # NOTE: we could try and detect cl.exe from the env vars below,
+        #       but Microsoft seems to change the location often.
+        #       so we just require the user to modify his PATH
+        # envvars = collect(filter(key->ismatch(r"^VS\d+.+TOOLS$", key), keys(ENV)))
```

Comment:
Those should still be fine for now, it's just that Visual studio 2017 changed things.

Reply:
let's just rely on `vswhere` for all versions?

Reply:
`vswhere` is only for 2017 and newer however, it doesn't seem to find my  VS 2015 install. 
Indeed looking at the description at  https://github.com/Microsoft/vswhere
"Locate Visual Studio 2017 and newer installations"

Reply:
Perhaps it would also just be better to use `%ProgramFiles(x86)%\Microsoft Visual Studio\Installer` instead of downloading the exe.

Reply:
Yeah, I thought it detected older versions hence the download. What a mess.

Reply:
I can confirm  it works with cuda 9 and vs studio 2017